### PR TITLE
Wabbit season

### PR DESCRIPTION
**Pull request by tank157**
Removes the use of rabbitmq from the application and replaces it with various AWS services (SQS mostly). A lot of refactoring and then tweaks to get it operating correctly in the staging environment. Tackled a number of configuration streamlines and version upgrades as well. Oh, and removed the not really used/useful newrelic integration as well.

[Pivotal card](https://www.pivotaltracker.com/story/show/166622758)

Pairs with branches of the same name in `data-processor` and `Metis`.

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -1,22 +1,24 @@
...
+  :dependencies [[org.clojure/clojure "1.10.1"]
                  [turbovote.resource-config "0.2.1"]
-                 [com.novemberain/langohr "4.0.0"]
-                 [org.clojure/tools.logging "0.4.0"]
+                 [org.clojure/tools.logging "0.5.0-alpha.1"]
+                 [org.clojure/data.json "0.2.6"]
                  [ch.qos.logback/logback-classic "1.2.3"]
-                 [amazonica "0.3.102"]
-                 [com.climate/clj-newrelic "0.2.1"]
+                 [com.amazonaws/aws-java-sdk-core "1.11.571"]
```
Comment (sduckett) [inline project.clj]: I think these are up to `.585` now

File: project.clj [hunk 2]
```diff
@@ -1,22 +1,24 @@
 (defproject batch-address-test-tool "0.1.0-SNAPSHOT"
   :description "An address verification tool"
-  :dependencies [[org.clojure/clojure "1.8.0"]
+  :dependencies [[org.clojure/clojure "1.10.1"]
                  [turbovote.resource-config "0.2.1"]
-                 [com.novemberain/langohr "4.0.0"]
-                 [org.clojure/tools.logging "0.4.0"]
+                 [org.clojure/tools.logging "0.5.0-alpha.1"]
+                 [org.clojure/data.json "0.2.6"]
```
Comment (sduckett) [inline project.clj:6]: looks like this is duplicated on line 19 below?

File: src/vip/batch_address_test_tool/processor.clj [hunk 3]
```diff
@@ -167,30 +167,26 @@
...
-                           "transactionId" (get-in ctx [:input "transactionId"])}
-                          "batch-address.file.complete"))
-    (let [response-message {"fileName" (get-in ctx [:results :file-name])
-                            "bucketName" (get-in ctx [:results :bucket-name])
-                            "status" "ok"
-                            "url" (get-in ctx [:results :url])
-                            "fipsCode" (get-in ctx [:input "fipsCode"])
-                            "transactionId" (get-in ctx [:input "transactionId"])}]
-      (q/publish-to-queue response-message "batch-address.file.complete"))))
+    (q/publish-failure {:status "error"
```
Comment (sduckett) [inline src/vip/batch_address_test_tool/processor.clj:172]: 👏 This reads better to my eyes.

File: src/vip/batch_address_test_tool/queue.clj [hunk 4]
```diff
@@ -1,82 +1,98 @@
...
+
+(defn publish-failure
+  "Publish a failed feed processing message to the topic."
+  ([payload]
+   (publish-failure (default-sns-client)
+                    (config [:aws :sns :address-failure-failure-arn])
+                    payload))
+  ([sns-client topic payload]
+   (let [json-payload (json/write-str payload)]
+     (log/info "publishing failure to " topic " with payload " json-payload)
```
Comment (sduckett) [inline src/vip/batch_address_test_tool/queue.clj]: You could use the format string version: `(log/infof "publishing failure to %s with payload %s" topic json-payload)`

**Review by sduckett — APPROVED** [review]
👍 🚫🐰
One less service to operate sounds pretty nice to me!

File: src/vip/batch_address_test_tool/queue.clj (hunk 4 above)
Comment (tank157) [reply src/vip/batch_address_test_tool/queue.clj]: I did not know about `log/infof`, thanks!

**Pull request merged**